### PR TITLE
Superuser from config only

### PR DESCRIPTION
**Pull request by andrewpatto**
Am doing an E2E walk through of a "blank" system.. and fixing things as I go.

Here, we want the users who are listed as super admins to be able to bootstrap themselves (and others) into having permissions that let us do more in the system.

### CODE REVIEW

File: application/backend/tests/service-tests/user.basic.test.ts [hunk 1]
```diff
@@ -65,19 +67,39 @@ it("upsert an existing user to a new display name", async () => {
 });
 
 it("SuperAdmin change other user permission", async () => {
-  const newUser = await userService.upsertUserForLogin(
+  const newContainer = testContainer.createChildContainer();
```
Comment (andrewpatto) [inline application/backend/tests/service-tests/user.basic.test.ts:70]: A possibly interesting example of how we can change the "settings" just for a single test

File: application/backend/src/business/db/user-queries.ts [hunk 2]
```diff
@@ -54,41 +54,3 @@ export const deletePotentialUserByEmailQuery = e.params(
...
-
-/**
- * A count of all users.
- */
-export const countAllUserQuery = e.count(e.permission.User);
-
-/**
- * A pageable EdgeDb query for all users.
- */
-export const pageableAllUserQuery = e.params(
```
Comment (andrewpatto) [inline application/backend/src/business/db/user-queries.ts:66]: converted into a real combined EdgeDb query

File: application/backend/src/api/api-auth-routes-test-user-helper.ts [hunk 3]
```diff
@@ -0,0 +1,156 @@
+import { FastifyInstance } from "fastify";
```
Comment (andrewpatto) [inline application/backend/src/api/api-auth-routes-test-user-helper.ts:1]: split out so the "non-test" portion of the login flow is more obvious - and easier to debug

File: application/backend/src/api/api-auth-routes.ts [hunk 4]
```diff
@@ -294,13 +173,16 @@ export const callbackRoutes = async (
 
     const idClaims = tokenSet.claims();
 
-    const displayName = idClaims.name || "No Display Name";
-    const email = idClaims.email || "No Email";
+    if (!idClaims.name || !idClaims.email) {
+      // TODO work out what we are doing about errors here
+      reply.redirect("/missing-name-or-email");
+      return;
+    }
```
Comment (andrewpatto) [inline application/backend/src/api/api-auth-routes.ts:180]: sorry @DoxasticFox - we probably want to disable ORCID log ins until we sort out how we can get emails etc
(given emails are a fundemantal to how people get invited into the release)

Comment (DoxasticFox) [reply application/backend/src/api/api-auth-routes.ts:180]: Not a problem for me. I only ever used `/dev-bm3ey56` for development

**Review by DoxasticFox — APPROVED** [review]

**Review by williamputraintan — APPROVED** [review]

**Pull request merged**